.github/workflows/solidity-foundry.yml

```diff
@@ -6,39 +6,102 @@ env:
   # Has to match the `make foundry` version in `contracts/GNUmakefile`
   FOUNDRY_VERSION: nightly-de33b6af53005037b463318d2628b5cfcaf39916
 
+# Making changes:
+# * use the top-level matrix to decide, which checks should run for each product.
+# * when enabling code coverage, remember to adjust the minimum code coverage as it's set to 98.5% by default.
+
 jobs:
+  define-matrix:
+    name: Define test matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.define-matrix.outputs.matrix }}
+    steps:
+      - name: Define test matrix
+        id: define-matrix
+        shell: bash
+        run: |
+          cat <<EOF > matrix.json
```

Comment:
maybe create a setting.json file?
https://www.jitsejan.com/use-github-actions-with-json-file-as-matrix

Reply:
I thought about it, but decided that locality of action is more important, so that you see everything in a single place